cvat/apps/consensus/merge_consensus_jobs.py

```diff
@@ -0,0 +1,282 @@
+from cvat.apps.consensus.new_intersect_merge import IntersectMerge
+from cvat.apps.dataset_manager.bindings import import_dm_annotations
+from cvat.apps.dataset_manager.task import PatchAction, patch_job_data
+from cvat.apps.engine.models import Job, JobType, StageChoice, StateChoice, Task, User
+from cvat.apps.engine.serializers import RqIdSerializer
+from cvat.apps.engine.utils import (
+    define_dependent_job,
+    get_rq_job_meta,
+    get_rq_lock_by_user,
+    process_failed_job,
+)
+from cvat.apps.quality_control.quality_reports import JobDataProvider
+
+
+class MergeConsensusJobs:
+    def __init__(self, task_id: int) -> None:
+        self.jobs: Dict[int, List[Tuple[int, User]]]
+        self.parent_jobs: List[Job]
+        self.merger: IntersectMerge
+        self.consensus_settings: ConsensusSettings
+        self.assignee_jobs_count: Dict[User, int]
+        self.task_id = task_id
+        self.get_consensus_jobs(task_id)
+        self.get_assignee_jobs_count()
+        self.consensus_settings = ConsensusSettings.objects.filter(task=task_id).first()
+        self.merger = IntersectMerge(
+            conf=IntersectMerge.Conf(
+                pairwise_dist=self.consensus_settings.iou_threshold,
+                output_conf_thresh=self.consensus_settings.agreement_score_threshold,
+                quorum=self.consensus_settings.quorum,
+                sigma=self.consensus_settings.sigma,
+                torso_r=self.consensus_settings.line_thickness,
+            )
+        )
+        if not self.jobs:
+            raise ValidationError(
+                "No annotated consensus jobs found or no regular jobs in annotation stage"
+            )
+
+    def get_consensus_jobs(self, task_id: int) -> None:
+        job_map = {}  # parent_job_id -> [(consensus_job_id, assignee)]
+        parent_jobs: dict[int, Job] = {}
+        for job in (
+            Job.objects.prefetch_related("segment", "parent_job", "assignee")
+            .filter(
+                segment__task_id=task_id,
+                type=JobType.CONSENSUS.value,
+                parent_job__stage=StageChoice.ANNOTATION.value,
+                parent_job__isnull=False,
+            )
+            .exclude(state=StateChoice.NEW.value)
+        ):
+            job_map.setdefault(job.parent_job_id, []).append((job.id, job.assignee))
+            parent_jobs.setdefault(job.parent_job_id, job.parent_job)
+
+        self.jobs = job_map
+        self.parent_jobs = list(parent_jobs.values())
+
+    def get_assignee_jobs_count(self) -> None:
+        assignee_jobs_count = {}
+        for assignees in self.jobs.values():
+            for _, assignee in assignees:
+                if assignee not in assignee_jobs_count:
+                    assignee_jobs_count[assignee] = 1
+                else:
+                    assignee_jobs_count[assignee] += 1
+        self.assignee_jobs_count = assignee_jobs_count
+
+    @staticmethod
+    def get_annotations(job_id: int) -> dm.Dataset:
```

Comment:
```suggestion
    def _get_annotations(job_id: int) -> dm.Dataset:
```
I don't think this function is supposed for use outside of this class.